MerlinAU.sh

```diff
@@ -606,75 +611,77 @@ ${REDct}v${SCRIPT_VERSION}${NOct} --> ${GRNct}v${DLRepoVersion}${NOct}"
    fi
 }
 
-##------------------------------------------##
-## Modified by ExtremeFiretop [2024-Jul-03] ##
-##------------------------------------------##
-#a function that provides a UI to check for script updates and allows you to install the latest version...
+##----------------------------------------##
+## Modified by Martinski W. [2024-Jul-03] ##
+##----------------------------------------##
 _SCRIPTUPDATE_()
 {
-   local ScriptFileDL="${ScriptFilePath}.DL"
+   local scriptVers  ScriptFileDL="${ScriptFilePath}.DL"
 
    _CheckForNewScriptUpdates_
-    if [ "$1" = "force" ]
-    then
-        echo -e "${CYANct}Force downloading latest version...${NOct}"
-        serverver=$(/usr/sbin/curl -LSs --retry 4 --retry-delay 5 "$SCRIPT_URL_BASE/version.txt")
-        echo -e "${CYANct}Downloading latest version ($serverver) of $SCRIPT_NAME${NOct}"
-        curl -LSs --retry 4 --retry-delay 5 "${SCRIPT_URL_BASE}/version.txt" -o "$SCRIPTVERPATH"
-        curl -LSs --retry 4 --retry-delay 5 "${SCRIPT_URL_BASE}/${SCRIPT_NAME}.sh" -o "$ScriptFileDL"
-        
-        if [ $? -eq 0 ] && [ -s "$ScriptFileDL" ]
-        then
-            mv -f "$ScriptFileDL" "$ScriptFilePath"
-            chmod 755 "$ScriptFilePath"
-            echo -e "${CYANct}$SCRIPT_NAME successfully updated${NOct}"
-            _ReleaseLock_
-            exec "$ScriptFilePath"
-        else
-            rm -f "$ScriptFileDL"
-            echo -e "${REDct}Download failed.${NOct}"
-        fi
-         return
-    fi
+
+   _DownloadScriptFiles_()
+   {
+      local retCode
+
+      curl -LSs --retry 4 --retry-delay 5 "${SCRIPT_URL_REPO}/version.txt" -o "$SCRIPTVERPATH"
+      curl -LSs --retry 4 --retry-delay 5 "${SCRIPT_URL_REPO}/${SCRIPT_NAME}.sh" -o "$ScriptFileDL"
+
+      if [ $? -eq 0 ] && [ -s "$ScriptFileDL" ]
+      then
+          mv -f "$ScriptFileDL" "$ScriptFilePath"
+          chmod 755 "$ScriptFilePath"
+          retCode=0
+      else
+          rm -f "$ScriptFileDL"
+          printf "\n${REDct}Download failed.${NOct}\n"
+          retCode=1
+      fi
+      return "$retCode"
+   }
+
+   if [ $# -gt 0 ] && [ "$1" = "force" ]
```

Comment:
Adjusting this `if [ "$1" = "force" ]` to this: `if [ $# -gt 0 ] && [ "$1" = "force" ]`
Is what ultimately resolved the unset $1 bug, but I didn't intend on using the force in the menu ever, just the regular update function.